packages/EasyCore/tests/Bridge/Symfony/Serializer/TrimStringsNormalizerTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyCore\Tests\Bridge\Symfony\Serializer;
+
+use EonX\EasyCore\Bridge\Symfony\Serializer\TrimStringsNormalizer;
+use EonX\EasyCore\Tests\Bridge\Symfony\AbstractSymfonyTestCase;
+use stdClass;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+final class TrimStringsNormalizerTest extends AbstractSymfonyTestCase
+{
+    /**
+     * @return mixed[]
+     *
+     * @see testSupportsDenormalizationSucceeds
+     */
+    public function dataProviderForSupportsDenormalization(): array
+    {
+        return [
+            'data is string' => [
+                '',
+                true,
+            ],
+            'data is array' => [
+                [],
+                true,
+            ],
+            'data is bool' => [
+                true,
+                false,
+            ],
+            'data is numeric' => [
+                123,
+                false,
+            ],
+            'data is object' => [
+                new stdClass(),
+                false,
+            ],
+        ];
+    }
+
+    /**
+     * @param mixed $data
+     *
+     * @dataProvider dataProviderForSupportsDenormalization
+     */
+    public function testSupportsDenormalizationSucceeds($data, bool $expectedResult): void
+    {
+        $decorated = $this->prophesize(DenormalizerInterface::class);
+        $normalizer = new TrimStringsNormalizer($decorated->reveal());
+
+        $result = $normalizer->supportsDenormalization($data, 'no-matter');
+
+        self::assertSame($expectedResult, $result);
+    }
+
+    /**
+     * @throws \Symfony\Component\Serializer\Exception\ExceptionInterface
+     */
+    public function testDenormalizeSucceeds(): void
+    {
+        $data = [
```

Comment:
Please, add a data provider with data and expected data. Add another case where pass just a string as a data.

Reply:
Done.